### PR TITLE
create a pre.requirements.txt file for new releases

### PR DESCRIPTION
**Pull request by McKnight-42**
Add `pre.requirements.txt` file for adapter for 1.7 pre

example pr: https://github.com/dbt-labs/dbt-core-bundles/pull/19/files

also removing 3.11 to make sure this works with dbt-spark support_python_versions.

### CODE REVIEW

**Review by QMalcolm — APPROVED** [review]

**Review by colin-rogers-dbt — APPROVED** [review]

File: release_creation/bundle/requirements/v1.7.pre.requirements.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+dbt-core~=1.7.0b1 --no-binary dbt-postgres
+dbt-snowflake~=1.7.0b1
```
Comment (colin-rogers-dbt) [inline release_creation/bundle/requirements/v1.7.pre.requirements.txt:2]: can we make these pull from 1.7.latest? similar to https://github.com/dbt-labs/dbt-core-bundles/blob/main/release_creation/bundle/requirements/v1.2.pre.requirements.txt

File: release_creation/bundle/requirements/v1.7.pre.requirements.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+dbt-bigquery~=1.7.0b1
+dbt-redshift~=1.7.0b1
+dbt-postgres~=1.7.0b1
+dbt-spark[PyHive,ODBC]~=1.7.0b1
+dbt-databricks~=1.7.0b1
+dbt-trino~=1.7.0b1
```
Comment (colin-rogers-dbt) [inline release_creation/bundle/requirements/v1.7.pre.requirements.txt:8]: we should comment out dbt-trino and databricks until they release a 1.7 beta

**Review by colin-rogers-dbt — CHANGES_REQUESTED** [review]
couple changes needed to the requirements file

File: release_creation/bundle/requirements/v1.7.pre.requirements.txt (hunk 1 above)
Comment (McKnight-42) [reply release_creation/bundle/requirements/v1.7.pre.requirements.txt:2]: Sure was curious if that will work if there is no 1.7.latest branch? this is a beta so all being done off main.

**Review by colin-rogers-dbt — APPROVED** [review]

**Pull request merged**